dataline-workers/src/main/java/io/dataline/workers/process/ProcessBuilderFactory.java

```diff
@@ -28,6 +28,12 @@
 
 public interface ProcessBuilderFactory {
 
-  ProcessBuilder create(Path jobPath, String imageName, String... args);
+  ProcessBuilder create(String imageName, String... args);
+
+  /**
+   * @param path a local filesystem path
+   * @return the path from the process's perspective
+   */
+  Path rebasePath(final Path path);
```

Comment:
I feel a little bit weird adding it here as opposed to just the `DockerProcessBuilderFactory`. However, if it's in the `DockerProcessBuilderFactory` it either needs to be static and accept the workspace root / mountSource as a parameter (which is hard since pbfs are passed through so many layers) or changing the pbf type to `DockerProcessBuilderFactory` everywhere it's being passed, which seems to break the abstraction.

My justification for this being here is that any type of process can be required to provide a way to access mounted files, even if it's an identity for something like a local non-containerized process.

Let me know if you disagree.

Reply:
You can keep it there. The whole PBF thing isn't the cleanest design. We will revisit in the future.